### ISSUE TITLE
`ClasswiseWrapper` double prefixes with `self.prefix` when enclosed in a `MetricCollection`

### ISSUE THREAD

**Issue by anivegesana** [Lightning-AI/torchmetrics#1915]
## 🐛 Bug

#1866 double prefixes with `self.prefix` when enclosed in a `MetricCollection`. This is because `MetricCollection` already handles prefixing, here:
https://github.com/Lightning-AI/torchmetrics/blob/a448ad3ff4329682a83fe1036ef21f35a2a8418a/src/torchmetrics/collections.py#L335-L339
but #1866 doesn't account for it.

### To Reproduce

Enclose a `ClasswiseWrapper` with a `prefix` within a `MetricCollection`.

<details>
  <summary>Finds a multiclass accuracy and a classwise f1 score.</summary>

```py
from torchmetrics import *
import torch

category_names = ['Tree', 'Bush']
num_classes = len(category_names)

input_ = torch.rand((5, num_classes, 3, 3))
target = torch.ones((5, num_classes, 3, 3)).long()


val_metrics = MetricCollection(
    {
        "accuracy": Accuracy(task="multiclass", num_classes=num_classes),
        "f1": ClasswiseWrapper(
            F1Score(
                task="multiclass",
                num_classes=num_classes,
                average="none",
                dist_sync_on_step=True,
            ),
            category_names,
            prefix="f_score_",
        ),
    },
    prefix="val/",
)

val_metrics["precision"](input_, target)
val_metrics(input_, target)
```

</details>

### Expected behavior

I should get `{'val/acc': tensor(0.), 'val/f1_Tree': tensor(0.), 'val/f1_Bush': tensor(0.)}`. I instead get `{'val/acc': tensor(0.), 'val/f1_f1_Tree': tensor(0.), 'val/f1_f1_Bush': tensor(0.)}`.

### Environment

- `torchmetrics` 1.0.0 via `pip`
- Python `3.10.6` & PyTorch `1.12`:
- OS: Linux

**Comment by anivegesana** [comment]
Perhaps adding a flag for `Metric` subclasses to specify if they want `MetricCollection`s to prefix them, like `_key_names_fix = False` can fix the issue.
